include/remill/Arch/Instruction.h

```diff
@@ -254,6 +257,148 @@ class Instruction {
     kCategoryConditionalAsyncHyperCall,
   } category;
 
+
+  class Flow {};
+
+  class DirectFlow : Flow {
+   public:
+    DirectFlow() = delete;
+    DirectFlow(uint64_t known_target, DecodingContext static_context);
+
+    uint64_t known_target;
+    DecodingContext static_context;
+
+    bool operator==(const DirectFlow &rhs) const;
+  };
+
+  class IndirectFlow : Flow {
+   public:
+    // We may have info in the decoder that tells us a context value
+    std::optional<DecodingContext> maybe_context;
+
+    IndirectFlow() = delete;
+    IndirectFlow(std::optional<DecodingContext> maybe_context);
+
+    bool operator==(const IndirectFlow &rhs) const;
+  };
+
+  class FallthroughFlow : Flow {
+   public:
+    FallthroughFlow(DecodingContext fallthrough_context);
+    FallthroughFlow() = delete;
+
+    DecodingContext fallthrough_context;
+
+    bool operator==(const FallthroughFlow &rhs) const;
+  };
+
+  class NormalInsn {
+   public:
+    NormalInsn() = delete;
+    NormalInsn(FallthroughFlow fallthrough);
+
+    FallthroughFlow fallthrough;
+    bool operator==(const NormalInsn &rhs) const;
+  };
+
+  class NoOp {
+   public:
+    NoOp() = delete;
+    NoOp(FallthroughFlow fallthrough);
+
+    bool operator==(const NoOp &rhs) const;
+
+    FallthroughFlow fallthrough;
+  };
+
+  class InvalidInsn {
+   public:
+    InvalidInsn() = default;
+
+    bool operator==(const InvalidInsn &rhs) const;
+  };
+
+  class ErrorInsn {
+   public:
+    ErrorInsn() = default;
+
+    bool operator==(const ErrorInsn &rhs) const;
+  };
+
+  class DirectJump {
+   public:
+    DirectJump() = delete;
+    DirectJump(DirectFlow taken_flow);
+
+    bool operator==(const DirectJump &rhs) const;
+    DirectFlow taken_flow;
+  };
+
+  class IndirectJump {
+   public:
+    IndirectJump() = delete;
+    IndirectJump(IndirectFlow taken_flow);
+
+    IndirectFlow taken_flow;
+
+    bool operator==(const IndirectJump &rhs) const;
+  };
+
+  class DirectFunctionCall : DirectJump {
+   public:
+    DirectFunctionCall() = delete;
+    DirectFunctionCall(DirectFlow taken_flow);
+
+    bool operator==(const DirectFunctionCall &rhs) const;
+  };
+
+
+  class IndirectFunctionCall : IndirectJump {
+   public:
+    IndirectFunctionCall() = delete;
+    IndirectFunctionCall(IndirectFlow id_flow);
```

Comment:
Copy & paste.
`using IndirectJump::IndirectJump;`

Reply:
I dont understand what you are suggesting here?